packages/core/src/modules/credentials/services/CredentialService.ts

```diff
@@ -743,8 +745,8 @@ export class CredentialService {
     )
 
     // Update record
-    credentialRecord.errorMsg = `${credentialProblemReportMessage.description.code}: ${credentialProblemReportMessage.description.en}`
-    await this.updateState(credentialRecord, CredentialState.None)
+    credentialRecord.errorMessage = `${credentialProblemReportMessage.description.code}: ${credentialProblemReportMessage.description.en}`
+    await this.updateState(credentialRecord, credentialRecord.state)
```

Comment:
Same here

Reply:
Hi @TimoGlastra, If we update the `updateState` with an `update`, it will be difficult to capture the event in the e2e test case?

https://github.com/hyperledger/aries-framework-javascript/blob/1041aa77bdc1c441d99e7852d4062766520d6597/packages/core/tests/credentials.test.ts#L483

Reply:
Yes, see my comment above:

> Also, with the None state being removed, there is no way to determine that you received a problem report, so we should probably look into adding some events related to problem reports in follow up PRs


We should probably add a new event (e.g. `ConnectionProblemReportReceived`, `CredentialProblemReportReceived` etc...) where the event looks something like:

```ts
{
   payload: {
     credentialRecord: { /* the record */ },
     problemReportMessage: { /* the problem report message */ }
}
}
```

Reply:
You can either add it to the current PR, or you can bypass it by listening for the `AgentMessageProcessedEvent` for now and filtering on the problem report @type

Reply:
@TimoGlastra , As of now I have removed the test case and I will add it into upcoming PR with the problem-report events